dask_glm/estimators.py

```diff
@@ -0,0 +1,137 @@
+
+        fit_kwargs = {'max_iter', 'tol', 'family'}
+
+        if solver == 'admm':
+            fit_kwargs.discard('tol')
+            fit_kwargs.update({
+                'regularizer', 'lamduh', 'rho', 'over_relax', 'abstol',
+                'reltol'
+            })
+        elif solver == 'proximal_grad':
+            fit_kwargs.update({'regularizer', 'lamduh'})
+
+        self._fit_kwargs = {k: getattr(self, k) for k in fit_kwargs}
+
+    def fit(self, X, y=None):
+        X_ = self._maybe_add_intercept(X)
+        self._coef = algorithms._solvers[self.solver](X_, y, **self._fit_kwargs)
+
+        if self.fit_intercept:
+            self.coef_ = self._coef[:-1]
+            self.intercept_ = self._coef[-1]
+        else:
+            self.coef_ = self._coef
+        return self
+
+    def _maybe_add_intercept(self, X):
+        if self.fit_intercept:
+            return add_intercept(X)
+        else:
+            return X
+
+
+class LogisticRegression(_GLM):
+    """
+    Parameters
+    ----------
+    fit_intercept : bool, default True
+        Specifies if a constant (a.k.a. bias or intercept) should be
+        added to the decision function.
+    solver : {'admm', 'gradient_descent', 'newton', 'bfgs', 'proximal_grad'}
+        Solver to use. See :ref:`algorithms` for details
+    regularizer : {'L1', 'L2'}
+        Regularizer to use. See :ref:`regularizers` for details.
+        Only used with ``admm`` and ``proximal_grad`` solvers.
+    max_iter : int, default 100
+        Maximum number of iterations taken for the solvers to converge
+    tol : float, default 1e-4
+        Tolerance for stopping criteria. Ignored for ``admm`` solver
+    lambduh : float, default 0.1
+        Only used with ``admm`` and ``proximal_grad`` solvers
+    rho, over_relax, abstol, reltol : float
+        Only used with the ``admm`` solver. See :ref:`algorithms.admm`
+        for details
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_classes, n_features)
+    intercept_ : float
+
+    Examples
+    --------
+    >>> from dask_glm.datasets import make_classification
+    >>> X, y = make_classification()
+    >>> lr = LogisticRegression()
+    >>> lr.fit(X, y)
+    >>> lr.predict(X)
+    >>> lr.predict_proba(X)
+    """
+
+    @property
+    def family(self):
+        return families.Logistic
+
+    def predict(self, X):
+        return self.predict_proba(X) > .5  # TODO: verify, multiclass broken
+
+    def predict_proba(self, X):
+        X_ = self._maybe_add_intercept(X)
+        return sigmoid(dot(X_, self._coef))
+
+    def score(self, X, y):
+        return accuracy_score(y, self.predict(X))
+
+
+class LinearRegression(_GLM):
+    @property
+    def family(self):
+        return families.Normal
+
+    def predict(self, X):
+        X_ = self._maybe_add_intercept(X)
+        return dot(X_, self._coef)
+
+    def score(self, X, y_true):
+        return mean_square_error(y, self.predict(X))
+
+
+def accuracy_score(y_true, y_pred):
```

Comment:
Should these functions be in:
- `utils` or
-  within the corresponding Estimator class
?